arch/armv7m/tls_internal.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * tls_internal.c
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/types.h>
+
+
+volatile uintptr_t arm_tls_ptr __attribute__((section("armtls"))) = 0;
+
+
+void aeabi_read_tp_internal(void)
```

Comment:
I think this function should be void *aeabi_read_tp_internal(void), as we're returning stuff from it. Not super critical, but it's somewhat erroneous that we're returning stuff from void function. C should save all registers apart from r1-r3 and r12, so adding r12 to push list should allow us to safely do mov %[temp], %[tls] and then return temp.